http/responseemitter.go

```diff
@@ -67,59 +69,50 @@ type responseEmitter struct {
 }
 
 func (re *responseEmitter) Emit(value interface{}) error {
-	ch, isChan := value.(<-chan interface{})
-	if !isChan {
-		ch, isChan = value.(chan interface{})
+
+	// Initially this library allowed commands to return errors by sending an
+	// error value along a stream. We removed that in favour of CloseWithError,
+	// so we want to make sure we catch situations where some code still uses the
+	// old error emitting semantics and _panic_ in those situations.
+	debug.AssertNotError(value)
+
+	// if we got a channel, instead emit values received on there.
+	if ch, ok := value.(chan interface{}); ok {
+		value = (<-chan interface{})(ch)
+	}
+	if ch, isChan := value.(<-chan interface{}); isChan {
+		return cmds.EmitChan(re, ch)
 	}
 
-	if isChan {
-		for value = range ch {
-			err := re.Emit(value)
-			if err != nil {
-				return err
-			}
-		}
-		return nil
+	re.once.Do(func() { re.preamble(value) })
+
+	re.l.Lock()
+	defer re.l.Unlock()
+	
+	if single, ok := value.(cmds.Single); ok {
+		value = single.Value
+		defer re.closeWithError(nil)
```

Comment:
Will throw away encoding errors.

Reply:
IMHO encoding errors should be passed to the caller instead of be used to close the stream. And `closeWithError` always returns `nil` so I figured we don't need it to handle the error.

Reply:
The issue is that we're closing the response stream as if the command succeeded when, really, the command failed.